browser/modules/AboutNewTab.jsm

```diff
@@ -370,8 +401,12 @@ let AboutNewTab = {
    *          The tile index from which the action came from.
    */
   reportSitesAction: function(message) {
-    let sites = NewTabUtils.links.getLinks().slice(0, message.data.length);
-    DirectoryLinksProvider.reportSitesAction(sites, message.data.action, message.data.index);
+    // Convert sites to objects.
+    for (let i = 0; i < message.data.sites.length; i++) {
```

Comment:
Maybe:

```JS
var parsedSites = message.data.sites.map(site => JSON.parse(site)); 
DirectoryLinksProvider.reportSitesAction(parsedSites, message.data.action, message.data.index);
```
Note also that JSON.parse can throw, but that's not handled. Might not be a problem so long as it's an invariant that a site is always parsable.